### ISSUE TITLE
Drag and Drop shapes

### ISSUE THREAD

**Issue by marr** [cracker0dks/whiteboard#24]
Hi, thanks for this tool. It works great! 
Would it be possible to allow the cursor tool to support drag and drop of shapes? So when clicking the select tool, I could select a shape that was drawn and move it on the board. This is different than the rectangular tool, in that it selects the actual shape, and not a rectangular selection around the shape.

Thank you

**Comment by cracker0dks** [comment]
TLDR: Hard task, can't do it at the moment sry.

Details:
This would be a very hard task since we need to find the shape first (this is not as easy as it sounds) If you draw a line (with the line tool), this is represented as two points in the background so if someone clicks somewhere on the canvas, how do we know that this line is meant or not? We would need to "follow the color" or something like that... or show the start and endpoint of lines so we can move them? But what about pen draws then? What is defined as shape anyway?
Maybe someone has a good idea and is motivated to implement, I can't do it atm, sry.

**Comment by marr** [comment]
Thanks for the response. What would you think about a feature that allowed for certain svgs to be brought into the canvas, like UML elements? Is that something you think would be possible? I guess I don't know the limitations of canvas, but with draw.io, there are shapes that can be dragged into the canvas and scaled and moved. Is that easier when the shapes are pre-defined?

**Comment by cracker0dks** [comment]
we could add predefined elements pretty easy but they would behave like an image you drop in right now. Ether drawn to the canvas or the background. Only the rotation is missing for this to work properly I guess. 

Workaround for now (If you don't need the rotation): You could make a folder with all the UML Elements you need (as images) and drop them in.
Greetings

**Comment by cracker0dks** [comment]
fyi: added function to rotate images
